packages/tale_shared/tests/test_retry.py

```
@@ -0,0 +1,51 @@
+        mock_pool.acquire.assert_awaited_once()
+        mock_pool.release.assert_awaited_once_with(mock_conn)
+
+    @pytest.mark.asyncio
+    async def test_releases_on_exception(self):
+        mock_conn = AsyncMock()
+        mock_pool = MagicMock(spec=asyncpg.Pool)
+        mock_pool.acquire = AsyncMock(return_value=mock_conn)
+        mock_pool.release = AsyncMock()
+
+        with pytest.raises(RuntimeError, match="test error"):
+            async with acquire_with_retry(mock_pool) as conn:
```

Ruff (0.15.2)

[error] Local variable `conn` is assigned to but never used

Remove assignment to unused variable `conn`

(F841)